packages/next-intl/src/middleware/middleware.test.tsx

```diff
@@ -2704,6 +2704,27 @@ describe('domain-based routing', () => {
         } satisfies Pathnames<ReadonlyArray<'en' | 'fr'>>
       });
 
+      it('redirects an unprefixed pathname from the fr locale on the en domain', () => {
+        middlewareWithPathnames(
+          createMockRequest('/a-propos', 'fr', 'http://en.example.com')
+        );
+        expect(MockedNextResponse.redirect).toHaveBeenCalled();
+        expect(MockedNextResponse.redirect.mock.calls[0][0].toString()).toBe(
+          'http://en.example.com/about'
+        );
+      });
```

Comment:
@markomitranic I've tried to recreate your case here.

See also the config above:

https://github.com/amannn/next-intl/blob/101bb5ee0874beafd08c31c4fa3b6596a9efd1e1/packages/next-intl/src/middleware/middleware.test.tsx#L2652-L2705

This seems to be working for me. A redirect to `/about` is currently added here for compatibility, e.g. when the user switches the domain without changing pathnames.

Can you help me find a failing test?